program/rust/src/time_machine_types.rs

```diff
@@ -35,13 +386,22 @@ pub struct PriceAccountWrapper {
 }
 impl PriceAccountWrapper {
     pub fn initialize_time_machine(&mut self) -> Result<(), OracleError> {
-        msg!("implement me");
-        Ok(())
+        self.time_machine.initialize()
     }
 
     pub fn add_price_to_time_machine(&mut self) -> Result<(), OracleError> {
-        msg!("implement me");
-        Ok(())
+        //skip if there are not two valid prices in the price account
+        if self.price_data.prev_timestamp_ == 0 {
+            return Ok(());
```

Comment:
Is this case really problematic, IMO it's just gonna be invalid anyway so we're fine.

Reply:
I think this is easier to read, but happy to replace it with the following comment:
```
//this would work on the first add_price as well, as the first entry is always invalid, so even though we would probabpy end up overflowing the first number we should be good
```

Reply:
I'm almost convinced this is ok.

Reply:
Replaced with comment!